Libplanet/Blockchain/BlockChain.cs

```diff
@@ -964,35 +970,31 @@ public IImmutableSet<TxId> GetStagedTransactionIds()
                 return new Tuple<long?, IReadOnlyList<BlockHash>>(null, new BlockHash[0]);
             }
 
-            BlockHash? branchpoint = FindBranchpoint(locator);
-            var branchpointIndex = branchpoint is { } h ? (int)Store.GetBlockIndex(h)! : 0;
+            if (!(FindBranchpoint(locator) is BlockHash branchpoint))
+            {
+                return new Tuple<long?, IReadOnlyList<BlockHash>>(null, new BlockHash[0]);
+            }
 
-            // FIXME: Currently, increasing count by one to satisfy
-            // the number defined by FindNextHashesChunkSize variable
-            // when branchPointIndex didn't indicate genesis block.
-            // Since branchPointIndex is same as the latest block of
-            // requesting peer.
-            if (branchpointIndex > 0)
+            if (!(Store.GetBlockIndex(branchpoint) is long branchpointIndex))
             {
-                count++;
+                return new Tuple<long?, IReadOnlyList<BlockHash>>(null, new BlockHash[0]);
             }
 
             var result = new List<BlockHash>();
-            foreach (BlockHash hash in Store.IterateIndexes(Id, branchpointIndex, count))
+            foreach (BlockHash hash in Store.IterateIndexes(Id, (int)branchpointIndex, count))
```

Comment:
It's a bit off-topic from this PR, but casting `long` block index into `int` seems odd.

Reply:
Eh, it is weird. We need to change `IStore` interface to accept `long` values at some point. 🙄